component/src/ibc/transfer.rs

```diff
@@ -28,6 +35,71 @@ impl ICS20Transfer {
     pub fn new(state: State) -> Self {
         Self { state }
     }
+
+    pub async fn withdrawal_check(&self, ctx: Context, withdrawal: &ICS20Withdrawal) -> Result<()> {
+        // create packet
+        let packet: IBCPacket<Unchecked> = withdrawal.clone().into();
+
+        // send packet
+        use crate::ibc::packet::SendPacket;
+        self.state.send_packet_check(ctx.clone(), packet).await?;
+
+        Ok(())
+    }
+
+    pub async fn withdrawal_execute(&mut self, ctx: Context, withdrawal: &ICS20Withdrawal) {
+        // create packet
+        let packet: IBCPacket<Unchecked> = withdrawal.clone().into();
```

Comment:
note: to avoid rechecking here or needing to communicate the withdrawal state from `withdrawal_check`, implement a `.assume_checked()` fn on IBCPacket that allows us to transform it to a Checked ibc packet without re-running all of the checks (which were already run in `withdrawal_check`) (thx @plaidfinch)